### ISSUE TITLE
typedef and 'for...of': type of left-hand side

### ISSUE THREAD

**Issue by Avol-V** [palantir/tslint#894]
I write this code, and TSLint gives me an error:
```typescript
let items: string[] = ['a', 'b'];
for ( let item of items ) // TSLint: expected variable declaration: 'item' to have a typedef (typedef)
{
	console.log( item );
}
```

OK, I write this, and TypeScript gives me an error:
```typescript
let items: string[] = ['a', 'b'];
for ( let item: string of items ) // TypeScript error: Error TS2483: The left-hand side of a 'for...of' statement cannot use a type annotation.
{
	console.log( item );
}
```

Looks like a bug?

**Comment by JKillian** [comment]
Hi @Avol-V, thanks for the bug report! I believe [this got fixed in the past](https://github.com/palantir/tslint/commit/a330ba043d64bb02cb97233fdd3852f5075218ff) when I was fixing #743.

What version of TSLint are you using?

**Comment by Avol-V** [comment]
Oh, yes, I'm sorry — I was looking for such issue, but only in opened.
Really it's already fixed.
I thought that I recently updated it, but it turned out that the version is outdated.
I update tslint and issue is gone.
Thanks.
